### PR TITLE
Additional migration documentation for relationship types

### PR DESCRIPTION
**Pull request by goneall**
Rather than removing support for the SPDX 2.X relationships, it looks like we can just reverse the direction and use one of the SPDX 3.X relationship types.

I also found one or two relationship types that was not documented.

### CODE REVIEW

**Comment by bact** [comment]
@goneall do you mind to fix typo hasPrequisite / hasPrerequisite typo at line 242 please?  So we can close #1017 and use this PR instead.

**Comment by goneall** [comment]
> It looks the direction column is not complete as well (not your PR but the existing one). I will just copied my comments from [#1017 (comment)](https://github.com/spdx/spdx-spec/pull/1017#issue-2440131935)
> 
> ### Questions
> 1. Does this table complete?
>    
>    * A sentence before the table said:
>      > Relationship migration is being worked out in the relationships spreadsheet. Once completed, the following table will reflect the translation for relationship types from SPDX 2.3 to SPDX 3.0:
> 2. What does "lifecycle scope" in the last column mean?
>    
>    * Other values in this column are types in LifecycleScopeType (for example, "build", "build, runtime", "test") or a string stating "various LifecycleScopeType".
>    * Should this "lifecycle scope" string be replaced by "various LifecycleScopeType"?
> 3. The 3rd column ("Swap to and from?") probably not complete?
>    
>    * For example, 2.3 `DATA_FILE_OF` and 3.0 `hasDataFile` they are swapped in direction. But there's no "Y" marked in their column.
>    * A (is a) `DATA_FILE_OF` B
>    * B `hasDataFile` A

We should fix these as well - I won't have time to go through another pass before my trip - perhaps we can open a separate PR for the remaining questions.

Ping @kestewart

**Review by bact — APPROVED** [review]

**Comment by bact** [comment]
> We should fix these as well - I won't have time to go through another pass before my trip - perhaps we can open a separate PR for the remaining questions.

Agree. Let's merge this first and manage other stuff in another PR.

**Comment by bact** [comment]
I have open PR #1022, as a patch to @goneall's branch, to add all missing directions, fix typos in descendantOf and hasVariant names, and address all the three questions above.

File: docs/annexes/diffs-from-previous-editions.md [hunk 1]
```diff
@@ -209,54 +209,55 @@ The completeness property would be constructed based on the following:
...
+| PATCH_APPLIED | patchedBy | Y | |
+| PREREQUISITE_FOR | hasPrerequisite | Y | various lifecycle scope |
+| PROVIDED_DEPENDENCY_OF | hasProvidedDependency | Y | various lifecycle scope |
+| REQUIREMENT_DESCRIPTION_FOR | hasRequirement | Y | various lifecycle scope |
+| RUNTIME_DEPENDENCY_OF | dependsOn | Y | runtime |
+| SPECIFICATION_FOR | hasSpecification | Y | various lifecycle scope |
+| STATIC_LINK | hasStaticLink | | various lifecycle scope |
+| TEST_CASE_OF | hasTestCase | Y | |
+| TEST_DEPENDENCY_OF | dependsOn | Y | test |
+| TEST_OF | hasTest | Y | lifecycle scope |
```
Comment (bact) [inline docs/annexes/diffs-from-previous-editions.md:258]:
```suggestion
| TEST_OF | hasTest | Y | various lifecycle scope |
```

I missed one here.

**Review by bact — CHANGES_REQUESTED** [review]

**Review by JPEWdev — APPROVED** [review]

File: docs/annexes/diffs-from-previous-editions.md (hunk 1 above)
Comment (bact) [reply docs/annexes/diffs-from-previous-editions.md:258]: @kestewart the other lifecycle-scoped rows in the table are already using this "various lifecycle scope" text.
This suggested change is to keep line 258 consistent with other rows.

**Review by bact — CHANGES_REQUESTED** [review]

File: docs/annexes/diffs-from-previous-editions.md (hunk 1 above)
Comment (bact) [reply docs/annexes/diffs-from-previous-editions.md:258]: @kestewart the other lifecycle-scope rows in the table already use the text "various lifecycle scope".

This suggested change is to keep Line 258 consistent with those other rows.

**Review by kestewart — APPROVED** [review]
Thanks for the cleanup and making things more consistent.

**Review by zvr — APPROVED** [review]

**Comment by bact** [comment]
To future readers, this doc is no longer in this repo and is now available at https://github.com/spdx/using/

**Pull request merged**